components/calendar/src/astronomy.rs

```diff
@@ -0,0 +1,657 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+/// This file contains important structs and functions relating to location,
+/// time, and astronomy; these are intended for calender calculations and based off
+/// _Calendrical Calculations_ by Reingold & Dershowitz.
+use crate::iso::Iso;
+use crate::rata_die::RataDie;
+use crate::types::Moment;
+use crate::{Date, Gregorian};
+
+#[derive(Debug)]
+pub struct Location {
+    latitude: f32,  // latitude from -90 to 90
+    longitude: f32, // longitude from -180 to 180
+    elevation: f32, // elevation in meters
+}
+
+// The mean synodic month in days of 86400 atomic seconds
```

Comment:
Ditto for here and the rest of the file. The CI doc job is ensuring that all public facing things have Rust doc comments (`///`). That way we don't have docs with lots of gaps, which is unhelpful for users, and makes everything less usable that way.